### DISCUSSION TITLE
HttpRequestRetryStrategy

### DISCUSSION THREAD

**Discussion by PJaroslaw** [keycloak/keycloak#23383] [Ideas]
I was wondering if it's planned to implement a `HttpRequestRetryStrategy` for requests made by keycloak.
E.g keycloak makes a call to a external service from `AbstractOAuth2IdentityProvider`, which results in a `NoHttpResponseException` or `SocketTimeoutExpcetion` or `SocketException` or any other `IOExpcetion`, then keycloak would retry those requests a given amount of times, maybe the default should be 1, but could be configurable.

**Comment by patrickbadleyupstart** [comment]
@PJaroslaw We are seeing a need for this as well.  Were you able to come up with a solution to this issue outside of this request?

**Reply by PJaroslaw** [reply]
Sadly no, we're just playing around with the `socket-timeout-millis` and `establish-connection-timeout-millis` but with no success tbh.

**Reply by PJaroslaw** [reply]
Just an update, that we haven't found a workaround for this yet outside of this request 😢
